extensions/support/policy/deployment/src/main/java/org/apache/camel/quarkus/component/support/policy/deployment/PolicyProcessor.java

```diff
@@ -0,0 +1,112 @@
+                "org.apache.camel.component.fhir.FhirConfiguration",
+                "org.apache.camel.component.fhir.FhirLoadPageEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirSearchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirTransactionEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCreateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirValidateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirReadEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCapabilitiesEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirHistoryEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirMetaEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirPatchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirDeleteEndpointConfiguration",
+
+                /* org.apache.camel.component.consul.* can be removed after the upgrade to Camel 3.1 */
+                "org.apache.camel.component.consul.ConsulConfiguration",
+                "org.apache.camel.component.consul.ConsulClientConfiguration",
+                "org.apache.camel.component.consul.health.ConsulHealthCheckRepositoryConfiguration",
+                "org.apache.camel.component.consul.cloud.ConsulServiceRegistryConfiguration");
+    }
+
+    /* Make the build fail as long as there are banned classes registered for reflection */
+    @BuildStep
+    void bannedReflectiveClasses(
+            CombinedIndexBuildItem combinedIndex,
+            List<ReflectiveClassBuildItem> reflectiveClasses,
+            List<UnbannedReflectiveBuildItem> unbannedReflectives,
+            BuildProducer<GeneratedResourceBuildItem> dummy // to force the execution of this method
+    ) {
+        final DotName uriParamsDotName = DotName.createSimple("org.apache.camel.spi.UriParams");
+
+        final Set<String> bannedClassNames = combinedIndex.getIndex()
+                .getAnnotations(uriParamsDotName)
+                .stream()
+                .filter(ai -> ai.target().kind() == Kind.CLASS)
+                .map(ai -> ai.target().asClass().name().toString())
+                .collect(Collectors.toSet());
+
+        final Set<String> unbannedClassNames = unbannedReflectives.stream()
+                .map(UnbannedReflectiveBuildItem::getClassNames)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
+        Set<String> violations = reflectiveClasses.stream()
+                .map(ReflectiveClassBuildItem::getClassNames)
+                .flatMap(Collection::stream)
+                .filter(cl -> !unbannedClassNames.contains(cl))
+                .filter(bannedClassNames::contains)
+                .collect(Collectors.toSet());
+
+        if (!violations.isEmpty()) {
+            throw new IllegalStateException(
```

Comment:
Should we really fail the build ?

Reply:
Yes, this is only with our tests. It won't happen at runtime (unless the user explictly depends on the new policy extension)